src/calendar/google/mod.rs

```diff
@@ -0,0 +1,364 @@
+pub mod models;
+use indexmap::IndexMap;
+use reqwest::header;
+use std::ops::Range;
+
+use chrono::{DateTime, SecondsFormat, Utc};
+use reqwest::{Request, Response};
+use reqwest_middleware::{ClientBuilder, ClientWithMiddleware, Middleware, Next};
+use task_local_extensions::Extensions;
+use tokio::sync::Mutex;
+
+/// Represents a timeframe with a start and end time
+pub type DateRange = Range<DateTime<Utc>>;
+
+/// Google calendar client for making requests to the google calendar api
+#[derive(Debug)]
+pub struct GoogleCalendarClient {
+    client: ClientWithMiddleware,
+    calendar_id: String,
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum ClientError {
+    /// Errpr while authenticating with google
+    #[error("Authentication error: {0}")]
+    GCloudAuthError(#[from] google_cloud_auth::error::Error),
+
+    /// Error while making a http request
+    #[error("RequestError: {0}")]
+    RequestError(#[from] reqwest::Error),
+
+    /// Error while executing some middleware code
+    #[error("RequestMiddlewareError: {0}")]
+    RequestMiddlewareError(#[from] reqwest_middleware::Error),
+
+    /// Error while building http headers
+    #[error("InvalidHeaderError: {0}")]
+    RequestInvalidHedaer(#[from] reqwest::header::InvalidHeaderValue),
```

Comment:
The enum variant has a typo in it. But I suggest to rename it to `InvalidHeaderValueError`.